### PR TITLE
Use platformdefines through lib instead of cpp

### PR DESCRIPTION
**Pull request by luqunl**
Include platformdefines.cpp doesn't seems a good practice. Compile platformdefines.cpp into platformdefines lib file

### CODE REVIEW

**Comment by luqunl** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Tests

**Comment by luqunl** [comment]
@dotnet-bot test all

**Comment by luqunl** [comment]
@dotnet-bot test this please

**Comment by luqunl** [comment]
@dotnet-bot test Windows_NT x64 Formatting please

File: tests/src/Common/Platform/platformdefines.h [hunk 1]
```diff
@@ -94,21 +94,21 @@ typedef WCHAR OLECHAR;
...
-error_t TP_putenv_s(LPTSTR name, LPTSTR value);
-void TP_ZeroMemory(LPVOID buffer, size_t sizeInBytes);
-error_t TP_itow_s(int num, LPWSTR buffer, size_t sizeInCharacters, int radix);
-LPWSTR TP_sstr(LPWSTR str, LPWSTR searchStr);
-LPSTR  HackyConvertToSTR(LPWSTR pwszInput);
-DWORD TP_CreateThread(THREAD_ID* tThread, LPTHREAD_START_ROUTINE worker,  LPVOID lpParameter);
-void TP_JoinThread(THREAD_ID tThread);
-void TP_DebugBreak();
-DWORD TP_GetFullPathName(LPWSTR fileName, DWORD nBufferLength, LPWSTR lpBuffer);
+DLL_EXPORT error_t TP_scpy_s(LPWSTR strDestination, size_t sizeInWords, LPCWSTR strSource);
```
Comment (AaronRobinsonMSFT) [inline tests/src/Common/Platform/platformdefines.h]: `DLL_EXPORT` shouldn't be needed here since this is for a static lib.

**Comment by luqunl** [comment]
@dotnet-bot test this please

**Comment by luqunl** [comment]
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test please

**Comment by luqunl** [comment]
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test  please

**Comment by luqunl** [comment]
@dotnet-bot  test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0) please

**Comment by luqunl** [comment]
@dotnet-bot help

**Comment by luqunl** [comment]
@dotnet-bot  test this please

**Comment by luqunl** [comment]
@dotnet-bot test Windows_NT x64 Formatting please

**Comment by luqunl** [comment]
@dotnet-bot test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness please
@dotnet-bot test Windows_NT x64 min_opt ryujit CoreCLR Perf Tests Correctness please
@dotnet-bot test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness please
@dotnet-bot test Windows_NT x86 min_opt ryujit CoreCLR Perf Tests Correctness please

**Comment by luqunl** [comment]
@dotnet-bot test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness please

**Comment by luqunl** [comment]
@dotnet-bot test this

**Comment by luqunl** [comment]
@dotnet-bot test this please

**Comment by luqunl** [comment]
These perf tests failure aren't related to this PR.

**Pull request merged**